### PR TITLE
Fixes URL redirection

### PR DESCRIPTION
**Pull request by falquaddoomi**
This PR introduces a mechanism to automatically redirect the user to the newest version of a variant if they request an older version of the variant. The flow goes something like this:

1. The user requests an old version of a variant, e.g. http://localhost:8080/variant/9712
2. Upon request, the version's ID is compared to the newest one and if it's different, the user is redirected to the ID of the new variant with a querystring parameter appended that indicates a redirection occurred (e.g., http://localhost:8080/variant/222415?redirectedFrom=9712).
 - Note that the redirect *replaces* the current page in the history, since if we went back we would be immediately redirected forward.
3. On reaching the current version's page, a notice is displayed at the top of the page mentioning the redirect, with a link to return to the originally-requested version.
4. If the user clicks the link, they are returned to the previous version with '?noRedirect=true' specified to prevent them from being redirected again. In that case, the page instead displays a warning that the variant is not the newest version, and provides a link to the newest one. (e.g., http://localhost:8080/variant/9712?noRedirect=true).

Outside sources that wish to link to older variants intentionally may suppress the out-of-date warning by appending *both* 'noRedirect=true' and 'noRedirectMsg=true' to the query. If only 'noRedirectMsg=true' is specified, the redirect will still occur.

In summary, here are the possible querystring parameter configurations relating to redirects:
- **/variant/[id]:** redirection occurs normally
- **/variant/[id]?redirectedFrom=[old_id]:** notice is displayed that a redirect occurred.
- **/variant/[id]?noRedirect=true:** requested version is displayed even if it's out of date, but with a warning.
- **/variant/[id]?noRedirect=true&noRedirectMsg=true:** the requested version is displayed, with no warning indicating if it's out of date.

### CODE REVIEW

**Comment by zfisch** [comment]
Code looks sound, testing also worked as I expected, though it would probably be best to let @amycoffin and @melissacline experiment with it on a sandbox to make sure it does what they would like and also to help finalize the design of the messaging (e.g. should "Previous versions of this variant" be something else like "All versions of this variant"?).

@falquaddoomi could you get a sandbox up and running for review by tomorrow's call?

**Comment by falquaddoomi** [comment]
Sure, it's up and running at http://brcaexchange-dev.gi.ucsc.edu:9005.

Here's an example of an old variant (ID 9172) that will immediately cause a redirect to its newest version (221570):
http://brcaexchange-dev.gi.ucsc.edu:9005/variant/9172

Here's the same variant with the noRedirect flag enabled, which should display a message about it being outdated:
http://brcaexchange-dev.gi.ucsc.edu:9005/variant/9172?noRedirect=true

The same outdated variant with both noRedirect and noRedirectMsg flags enabled, causing no message to be displayed:
http://brcaexchange-dev.gi.ucsc.edu:9005/variant/9172?noRedirect=true&noRedirectMsg=true

**Comment by amycoffin** [comment]
@falquaddoomi here are my recommendations for text tweaks. @maryjgoldman do you have opinions?

http://brcaexchange-dev.gi.ucsc.edu:9005/variant/221570?redirectedFrom=9172
- You are viewing the most recent data on this variant.
  - The variant url you requested is from an old release. You have been automatically redirected to updated data. Click here to view variant data from Release 1 (06 October 2016). 

http://brcaexchange-dev.gi.ucsc.edu:9005/variant/9172?noRedirect=true

- There is new data on this variant available
  - The data below is from Release 1 (06 October 2016). Click here for updated data on this variant.

**Comment by zfisch** [comment]
@falquaddoomi I think the other request was a change from `Previous Versions of this Variant (up to 12 April 2019)` to `Variant History`. Does that sound right @amycoffin @melissacline ?

**Comment by falquaddoomi** [comment]
@zfisch About that, should I remove the '(up to XXX)' bit, then? I suppose it's obvious what it's up to, since it's the same as the most recent element in the history.

**Comment by amycoffin** [comment]
@falquaddoomi @zfisch Yes, that looks right. I would probably remove the "up to" phrase myself, since I think it is obvious enough (especially given the GC use-case of this feature).

**Comment by maryjgoldman** [comment]
Yes, I was going to say that I thought the '(up to XX)' was a hair confusing so 'Variant history' plain and simple seems best.

Small edit to the old variant page:
'There is new data available on this variant' --> a bit easier to read out loud.

**Comment by maryjgoldman** [comment]
I'm also thinking about the word 'release' and how it's a bit technical. I made some edits below to put the emphasis on the date rather than the release. Any thoughts? The first one seems a bit awkward but maybe that's just me. @amycoffin 

The variant url you requested only has data up to 06 October 2016. You have been automatically redirected to the newest data. Click here to view variant data from 06 October 2016 (Release 1).

The data below is from 06 October 2016 (Release 1). Click here for updated data on this variant.

**Comment by amycoffin** [comment]
@maryjgoldman I like your wording!

**Comment by zfisch** [comment]
@falquaddoomi go ahead and merge as soon as the final updates are made :)

nice work and thanks everybody for quick and thorough feedback on wording!

**Comment by zfisch** [comment]
I actually think this has been updated to the latest comments, if any further changes are needed we can address them before going to production.. merging.
